deploy.sh

```diff
@@ -1,10 +1,132 @@
+  file_list+=( ["solr/data/solr_data.tgz"]="https://reactome.org/download/current/solr_data.tgz" ) # solr data
+  file_list+=( ["java-application-builder/downloads/analysis_v61.bin.gz"]="https://reactome.org/download/current/analysis_v61.bin.gz" ) # Analysis.bin for analysis service
+  file_list+=( ["java-application-builder/downloads/interactors.db.gz"]="https://reactome.org/download/current/interactors.db.gz" ) # interactors.db required to create analysis.bin
+  file_list+=( ["java-application-builder/downloads/diagrams_and_fireworks.tgz"]="https://reactome.org/download/current/diagrams_and_fireworks.tgz" )
+  # file_list+=( ["mysql/wordpress_data/reactome-wordpress.sql.gz"]="http://www.reactome.org/download/current/databases/gk_wordpress.sql.gz")
+
+  for db_file in "${!file_list[@]}";
+  do
+    # Initialization before prepairing download
+    URL=${file_list[${db_file}]}
+    file_path=${db_file}
+    file_name=$(basename $file_path)
+    mkdir -p $(dirname $file_path)
+
+    # Get size information
+    typeset -i remote_file_size=$(curl -sI $URL | tr -d '\r' | grep -i content-length | awk '{print $2}')
+    typeset -i local_file_size=$(stat -c %s -- $file_path)
+    echo "======================================================================="
+    echo "======================================================================="
+    echo "Filename:    " $file_name
+    echo "Remote Size: " $remote_file_size
+    echo "Local Size:  " $local_file_size
+
+    if [[ $local_file_size -eq $remote_file_size ]]; then
+      echo "Database up to date. Update not required"
+    elif [[ $remote_file_size -eq 0 ]]; then
+      echo "Remote file not acccessible. Not updating!"
+    else
+      echo "Database needs to be updated!"
+      echo "Removing old file if it exists!"
+      rm $file_path 2> /dev/null # 2> /dev/null is to ignore error if file not found
+      echo "Downloading newer version"
+      # To resume partially completed download, use --continue flag and comment out "rm $file_path 2> /dev/null"
+      wget -O $file_path $URL
+    fi
+    echo
+    echo
+  done
+else
+    echo "No internet access! Not verifying databases!"
+fi
+
+
+echo -e "\n\n"
+echo "==========================================================================="
+echo "                           Unpacking required files"
+echo "==========================================================================="
+if [[ ! -f solr/data/solr_data_extracted.flag ]]; then
+  echo "Unpacking SolrData"
+  rm -rf solr/solr_data
+  tar -xvzf solr/data/solr_data.tgz -C solr/data
+  touch solr/data/solr_data_extracted.flag
+  chmod a+w solr/data/solr_data/reactome
+  chmod a+w solr/data/solr_data/reactome/data
+  chmod a+w solr/data/solr_data/reactome/data/index/write.lock
+  chmod a+w solr/data/solr_data/reactome/data/index
+  chmod a+w -R solr/data/solr_data/reactome/data/tlog
```

Comment:
I think its crucial how docker is configured on the users system. While I was looking for logging in mysql, I realized that there might be permission differences between host user and container user. 

 When we map from a container to the host, no mappings are made on user id's, and the name attached to the uid inside the container may be very different from outside. We need to initialize the directory permissions with something the container user can write to. One simple method is to create a group that has access to write to the files on both the host and container, and then add the various users to this group on both your image and host OS. As in my case, I had added my current user to DOCKER group.

I think we need to keep this thing in mind while preparing the documentation for installation and setup of this project on any new system.